etl/steps/data/meadow/cedlas/2024-03-08/sedlac.py

```diff
@@ -0,0 +1,299 @@
+        inequality_deciles_indices_tables, COUNTRIES, DECILES_INDICES_COLUMNS
+    )
+    inequality_gini_tables = format_long_tables(inequality_gini_tables, COUNTRIES, GINI_COLUMNS)
+
+    poverty_tables = format_long_tables(poverty_tables, COUNTRIES, POVERTY_COLUMNS, POVERTY_COLUMNS_TO_DROP)
+
+    #
+    # Save outputs.
+    #
+    # Create a new meadow dataset with the same metadata as the snapshot.
+    ds_meadow = create_dataset(
+        dest_dir,
+        tables=inequality_deciles_indices_tables + inequality_gini_tables + poverty_tables,
+        check_variables_metadata=True,
+        default_metadata=snap_poverty.metadata,
+    )
+
+    # Save changes in the
+    # new meadow dataset.
+    ds_meadow.save()
+
+
+def load_tables_from_snapshot(snap: Snapshot, sheets: Dict) -> List[Table]:
+    """Load all the sheets from the snapshot."""
+    tables = []
+    for sheet in sheets:
+        tb = snap.read(sheet_name=sheet, header=sheets[sheet]["header"])
+        tb.metadata.short_name = sheets[sheet]["short_name"]
+
+        tables.append(tb)
+    return tables
+
+
+def format_long_tables(
+    tb: List[Table], countries: List[str], columns: Dict, cols_to_drop: List[str] = []
+) -> List[Table]:
+    """Format inequality and poverty tables, which share a similar, long format."""
+
+    tables = []
+    for t in tb:
+        # Flatten column names
+        t.columns = ["_".join(map(str, x)) if isinstance(x, tuple) else str(x) for x in t.columns.values]
+
+        # Rename first columm to "index", to facilitate the next steps
+        t = t.rename(columns={t.columns[0]: "index"})
+
+        # Strip spaces from first column
+        t["index"] = t["index"].astype(str)
+        t["index"] = t["index"].str.strip()
+
+        # Create column country by selecting the rows in countries list
+        t["country"] = t.loc[t["index"].isin(countries), "index"]
+        t["country"] = t["country"].fillna(method="ffill")
+
+        # Do the same for year
+        t["year"] = t.loc[t["index"].astype(str).str[:4].str.isnumeric(), "index"]
+
+        # Assert if null values in index are for for certain countries
+        if t.m.short_name in ["ineq_deciles_ni", "ineq_indices_ni"]:
+            countries_with_null = ["Argentina", "Chile", "Mexico"]
+        elif t.m.short_name == "gini3":
+            countries_with_null = ["Argentina", "Brazil", "Chile"]
+        else:
+            countries_with_null = ["Argentina", "Chile"]
+
+        assert (
+            t[t["index"] == "nan"]["country"].unique().tolist() == countries_with_null
+        ), f"Null values in index are not only for {countries_with_null}."
+
+        # Assert that the null values are only two
+        assert len(t[t["index"] == "nan"]) == len(
+            countries_with_null
+        ), f"There are more than {len(countries_with_null)} null values in index."
+
+        # Replace empty index values for country = Chile and Argentina and delete for Brazil and Mexico
```

Comment:
it's a bit hard to quickly understand the reasons for some of these lines of code  but the dataset looks like such a mess so it's already super impressive you've gone this far to document various steps! :)

Reply:
Thanks, yeah, it's because they identify survey spells by assigning a name and sometimes there is a jump and just a Excel comment. I am adding more text